### PR TITLE
Issue 192 execute dag restarts

### PR DESCRIPTION
**Pull request by richardjgowers**

### CODE REVIEW

File: gufe/protocols/protocoldag.py [hunk 1]
```diff
@@ -359,6 +359,7 @@ def execute_DAG(protocoldag: ProtocolDAG, *,
                 keep_shared: bool = False,
                 keep_scratch: bool = False,
                 raise_error: bool = True,
+                n_retries: int = 0,
```
Comment (IAlibay) [inline gufe/protocols/protocoldag.py:362]: Are we changing the default in quickrun?

Comment (richardjgowers) [reply gufe/protocols/protocoldag.py:362]: Yes, different repo but 2 retries would make sense?

Comment (IAlibay) [reply gufe/protocols/protocoldag.py:362]: Yeah my question was more "here or there".

**Review by dwhswenson — APPROVED** [review]
LGTM, although I'm not sure what's going on with the existing CI failure.

A lot of the details here will change when we integrate #186 into `execute_DAG`, but I don't see anything that will cause fundamental conflicts.

**Pull request merged**